### ISSUE TITLE
code

### ISSUE THREAD

**Issue by lclszsdnr** [linhuixiao/HiVG#14]
您好，我在测试时出现了clip模型结构和权重中结构不一致的问题，请问您有什么建议吗
运行的 命令为eval_single_dataset_finetuning_base.sh 下的 对unc 数据集的测试，
具体如下，CUDA_VISIBLE_DEVICES=0,1 python -m torch.distributed.launch --nproc_per_node=2 --master_port 28888 --use_env hivg_eval.py --num_workers 2 --batch_size 60  --dataset unc           --vl_hidden_dim 512 --imsize 224 --max_query_len 77 --normalize_before --enable_adaptive_weights --use_mask_loss  --save_hilora_clip --hi_lora_stage 3 --data_root /ssh/lcl/hivg/data/image_data --split_root /ssh/lcl/hivg/data/ref_data_shuffled --eval_model /ssh/lcl/hivg/pretrain_weight/finetuning_base/unc/best_checkpoint.pth      --eval_set val    --output_dir /ssh/lcl/hivg/pretrain_weight/finetuning_base/unc;

此外修改了hivg.py下clip 下加载时的 self.clip = CLIPModel.from_pretrained("/ssh/lcl/transformer/clip/base16")，这个权重是从transformer库下https://huggingface.co/openai/clip-vit-base-patch16下载的。
本来是没有报错的，但是准确率很低，我将hivg_eval.py 文件中     missing_keys, unexpected_keys = model_without_ddp.load_state_dict(checkpoint['model'], strict=True) strict 改为 true 后就显示 结构不一致的错误，

报错地方在于 模型代码里结构为 text_model.encoder.layers.0.self_attn.k_proj.base_layer.weight，带有base_layer, 但是加载的您给的权重文件中没有，显示text_model.encoder.layers.0.self_attn.k_proj.weight.

**Comment by lclszsdnr** [comment]
我可以修改加载权重中的名称来解决此问题，但是想请教一下您该问题是因为什么导致的

**Comment by linhuixiao** [comment]
@lclszsdnr     Hi, did you reproduce the results? In addition, which version of the PEFT library do you have? Speculation is that the issue may be caused by the wrong version of the PEFT library. Please check the versions of the main libraries (peft, transformers) in my requirements.txt and re-configure the environment, and the issue may be resolved.

Also, try downloading the original CLIP model from the link I provided, I'm not sure whether Huggingface's model has been updated.

**Comment by lclszsdnr** [comment]
感谢您的回复，确实是因为PEFT版本问题导致的。
不过结果如您论文图中有些许出入，请问结果是正常 的吗，在unc val 中有86.7与论文结果相比下降，在 unc+中有68.24与论文结果相比有上升。

**Comment by linhuixiao** [comment]
@lclszsdnr    A little bit of difference is normal. Also, the hyperparameter settings provided by this repository are not for the best-performing models. Therefore, if you want to go further with HiVG, you can tune the hyperparameters and observe a clear performance improvement.
